ConcurrentAPI.adoc

```diff
@@ -0,0 +1,220 @@
+== Distributed Concurrent API
+
+=== Motivation
+
+Coming from an initial issue where we need to provide an implementation for vert.x cluster manager for the shared data feature, provide concurrent and distributed programming capabilities.
+
+* Distributed locks -> is it a good idea ?
+* Distributed semaphores : first use case, vert-x cluster management
+* Distributed Countdown latch : it seems the team has already spoken about having it
+* Distributed Atomic Variables : we already have them, would be worth to offer an specific AtomicLong (for example) ?
```

Comment:
I was looking into these atomic objects implementation and woooh... It uses cache.put() calls and event listeners basically as a multicast messaging API, with guaranteed delivery order.

It does not keep the actual values in the cache, user code is expected to hold(or lose) all replicas. So contrary to the blogpost, I don't think that this is
> making transparently the object replicated and durable

(speaking about durability: this does not faciliate cache stores)

While I could ignore that listeners might not be 100% reliable and the code does not handle retries, the 'responses' are always multicast as well. It's simply wrong tool for the job.